CONTRIBUTING.md

```diff
@@ -1,35 +1,77 @@
+> reviewers to otherwise get stuck on.
+> To make your (and our own) life easier, we provide a
+> [`clang-format` configuration file](https://github.com/root-project/root/blob/master/.clang-format).
 
 By providing code, you agree to transfer your copyright on the code to the "ROOT project".
 Of course you will be duly credited: for sizable contributions your name will appear in the
-[CREDITS](https://raw.githubusercontent.com/root-project/root/master/README/CREDITS){:target="_blank"}
+[CREDITS](https://raw.githubusercontent.com/root-project/root/master/README/CREDITS)
 file shipped with every binary and source distribution.
 The copyright transfer helps us with effectively defending the project in case of litigation.
 
-:warning: We require PRs to cleanly apply to master without a merge commit, i.e. through "fast-forward".
-Please follow the [coding conventions](https://root.cern.ch/coding-conventions),
-as this is a simple item for reviewers to otherwise get stuck on.
+## Your Commit
+
+Each commit is a self-contained, _atomic_ change. This means that:
+1. **Each commit should be able to successfully build ROOT.**
+Doing so makes traveling through the git history, for example during a `git bisect` much easier.
+Ideally, the commit also should not depend on other commits to _run_ ROOT.
+2. **Each commit does not contain more than one independent change.**
+This allows us to revert changes when needed, without affecting anything else.
+
+> [!TIP]
+> During a code review, it may be useful to make smaller commits to track intermediate changes, and rebase after the PR
+> is approved to ensure the above points are met and to reduce clutter.
+
+### Your Commit Message
+
+The commit summary (i.e. the first line of the commit message) should be preceded by the part of ROOT that is affected
+by your commit, in square brackets. See the [commit log](https://github.com/root-project/root/commits/master/) for
+examples. If you're not sure which scope you change belongs to, don't hesitate to ask! The summary itself should not
+exceed 50 characters, be meaningful (i.e., it describes the change) and should be written in the
+[present imperative mood](https://git.kernel.org/pub/scm/git/git.git/tree/Documentation/SubmittingPatches?id=HEAD#n239)
+(e.g. `Add this awesome feature` instead of `Adds this awesome feature` or `Added this awesome feature`).
+
+The commit message that follow the summary can be used to provide more context to the change.
+It should describe the **why**, rather than the **what** and **how** (we can gather this from the commit summary and the
+change diff, respectively).
+The commit message should be wrapped at 72 characters.
+
+## Your Pull Request
+
+> [!NOTE]
+> For the mechanics on how to create pull requests, please visit
+> [this page](https://root.cern/for_developers/creating_pr).
+
+The title of your PR follows the same principle as the commit summary. If your PR only involves one commit, you can
```

Comment:
Does the 50 chars limit hold also for PR titles?

Reply:
Good question, I don't really have a strong opinion on this but I think in principle we could be a bit more lenient here (maybe use the 72 chars from the commit message) since they are usually only viewed in the web interface.